iceoryx_hoofs/source/posix_wrapper/shared_memory_object.cpp

```diff
@@ -93,10 +93,26 @@ expected<SharedMemoryObject, SharedMemoryObjectError> SharedMemoryObjectBuilder:
         return error<SharedMemoryObjectError>(SharedMemoryObjectError::SHARED_MEMORY_CREATION_FAILED);
     }
 
+    const auto real_size = sharedMemory->get_size();
+    if (!real_size)
+    {
+        printErrorDetails();
+        IOX_LOG(ERROR) << "Unable to create SharedMemoryObject since we could not acquire the memory size of the "
+                          "underlying object.";
+        return error<SharedMemoryObjectError>(SharedMemoryObjectError::UNABLE_TO_VERIFY_MEMORY_SIZE);
+    }
+
+    if (*real_size < m_memorySizeInBytes)
+    {
+        printErrorDetails();
+        IOX_LOG(ERROR) << "Unable to create SharedMemoryObject since it does not fulfill the size requirements.";
+        return error<SharedMemoryObjectError>(SharedMemoryObjectError::SMALLER_THAN_MIN_REQUESTED_SIZE);
```

Comment:
I had to wrap my head around the name. What is smaller than the min requested size? How about `REQUESTED_SIZE_EXCEEDS_ACTUAL_SIZE`?